extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusSecurityIdentity.java

```diff
@@ -38,7 +38,7 @@ public Principal getPrincipal() {
 
     @Override
     public boolean isAnonymous() {
-        return false;
+        return principal == null || principal.getName() == null || principal.getName().trim().isEmpty();
```

Comment:
I think trim() is a bit of an overkill but isEmpty() makes sense. All principal names will need to be trimmed on every call to isAnonymous...

Reply:
@renegrob I don't mind, if I remove `trim` then one of the tests will pass, I'd use `isBlank` but we still support Java 8 :-)